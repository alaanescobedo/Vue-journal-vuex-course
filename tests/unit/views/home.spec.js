import HomeView from '@/views/Home';
import { shallowMount } from '@vue/test-utils';

describe('Home View', () => {
  let wrapper;
  const mockRouter = {
    push: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(HomeView, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });
  });

  test('Debe renderizar el componente correctamente', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Al hacer click el boton debe de redireccionar a no-entry', () => {
    wrapper.find('button').trigger('click');

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'login' });
  });
  test('Al hacer click el boton debe de redireccionar al login', () => {
    const buttons = wrapper.findAll('button');
    buttons[1].trigger('click');

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
  });
});
