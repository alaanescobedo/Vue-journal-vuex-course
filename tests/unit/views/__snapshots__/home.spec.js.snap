// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Debe renderizar el componente correctamente 1`] = `<div class="home"><img alt="Vue logo" src=""></div><button class="btn btn-primary">Login</button><button class="btn btn-primary">Daybook</button>`;
